interactions/models/internal/extension.py

```diff
@@ -128,6 +129,11 @@ def __new__(cls, bot: "Client", *args, **kwargs) -> "Extension":
             else:
                 raise TypeError("async_start is a reserved method and must be a coroutine")
 
+        instance.load(*args, **kwargs)
+        with contextlib.suppress(asyncio.CancelledError):
```

Comment:
I very much dislike us suppressing this, we're ignoring potential cancellation and suspension of async call sites when we're specifically wanting to determine sync/async behaviour. I think only one of these states should be allowed, depending on the validity of the guard condition.

Reply:
Was supposed to be `RuntimeError` - i guess that was removed during a rebase locally. 

Regardless, marking as resolved because im going to switch this out and only handle the loop check